### DISCUSSION TITLE
Error installing plugin mongodb *

### DISCUSSION THREAD

**Discussion by amolngt** [apache/skywalking#12878] [Q&A]
Hi all,

I am using below versions -

node - v16.14.2
mongo db version - 4.4.25
mongodb (npm package) - 4.4.1
skywalking-backend-js (npm package) - tried both 0.4.1 & 0.7.0

getting error below when i do npm start(load my nodejs app.)
```
{
  message: 'Error installing plugin mongodb *',
  level: 'error',
  file: '/repo_name/node_modules/skywalking-backend-js/lib/core/PluginInstaller.js'
}
```

Thank you in advance.

**Comment by wu-sheng** [comment]
You should check test matrix, whether your version is tested.

**Reply by amolngt** [reply]
how to do that?

**Reply by wu-sheng** [reply]
@kezhenxu94 I think we only test few versions?

**Reply by amolngt** [reply]
@wu-sheng can you tell me which version you have tested?

**Reply by wu-sheng** [reply]
I can find this, https://github.com/apache/skywalking-nodejs/pull/44

**Reply by amolngt** [reply]
not helpful.

**Reply by wu-sheng** [reply]
Isn't it writing 5.12 mongoose? The client lib version matters.

**Reply by amolngt** [reply]
Actually sir, i am not using mongoose. i am directly using MongoClient getting from mongo npm package having version 4.4.1.

**Reply by wu-sheng** [reply]
That was older, https://github.com/apache/skywalking-nodejs/pull/33

Version 3.6.4

**Reply by amolngt** [reply]
Can you tell me which version or any changes should i do, which will support my version 
node - v16.14.2
mongo db version - 4.4.25
mongodb (npm package) - 4.4.1

?

**Reply by wu-sheng** [reply]
No, there's not.
Any new client version support requires someone adding new adoption codes.
There is no magic here. Anything about auto is someone did that for you.

**Comment by amolngt** [comment]
can u tell me here for which node , mongo db version, mongodb (npm package) will this work ?

**Comment by kezhenxu94** [comment]
The tested version is 3.6 https://github.com/apache/skywalking-nodejs/blob/fa7fd652d7a527c457efb99429ca03ddf23b370f/package.json#L64
